routes/weather.ts

```diff
@@ -207,18 +209,22 @@ export const getWateringData = async function( req: express.Request, res: expres
 		// Reconstruct JSON string from deformed controller output
 		adjustmentOptions = JSON.parse( "{" + adjustmentOptionsString + "}" );
 	} catch ( err ) {
-
-		// If the JSON is not valid then abort the claculation
-		res.send(`Error: Unable to parse options (${err})`);
+		// If the JSON is not valid then abort the calculation
+		sendWateringData( res, { errCode: ErrorCode.MalformedAdjustmentOptions, errMessage: `Unable to parse options (${ err })` } );
 		return;
 	}
 
 	// Attempt to resolve provided location to GPS coordinates.
 	let coordinates: GeoCoordinates;
 	try {
 		coordinates = await resolveCoordinates( location );
-	} catch (err) {
-		res.send(`Error: Unable to resolve location (${err})`);
+	} catch ( err ) {
+		let codedError: CodedError = makeCodedError( err );
+		if ( codedError.errCode === ErrorCode.UnexpectedError ) {
+			console.error( `An unexpected error occurred during location resolution for "${ req.url }": `, err );
+		}
+
+		sendWateringData( res, { errCode: codedError.errCode, errMessage: `Unable to resolve location "${ location }" (${ codedError.message })` } );
```

Comment:
Might be worth having a `sendWateringError()` and pass in `err`. That way you can dedupe some of the boilerplate and also add things like dump stack trace to log if `UnexpectedError/InternalError` rather than console.log.